### ISSUE TITLE
settings about semi-supervised DA

### ISSUE THREAD

**Issue by lyxok1** [tim-learn/ATDOC#1]
Thanks for sharing the code~ After reading the codebase, I find the provided demo is aimed at unsupervised DA, while your paper also provide some promising results on SSDA setting. Therefore I wonder if there are any difference between hyperparameter settings under DA and SSDA scenario. 

For example

- the training batch distribution for label-source, label-target and unlabel-target 
- whether the labeled target feature & score are used to update memory
- the scheme to adjust weight $lambda$ of self-supervised loss term
- whehter the hyperparameters are different for other datasets like DomainNet

BTW, I find the following code for NA classifier (Line 305 from demo_uda.py) seems not consistent with the Eq (6) in paper

`
outputs_target = softmax_out**2 / ((softmax_out**2).sum(dim=0))
`

From Eq (6) in the paper, the normalization is over prediction on all classes to make the distribution sharper, while in the code the parameter is `dim=0`, resulting normalization over all unlabeled data samples, will the normalization direction make big difference in final results ?

Appreciation if you can provide more detail about it~

**Comment by tim-learn** [comment]
> Thanks for sharing the code~ After reading the codebase, I find the provided demo is aimed at unsupervised DA, while your paper also provide some promising results on SSDA setting. Therefore I wonder if there are any difference between hyperparameter settings under DA and SSDA scenario.
> 
> For example
> 
> * the training batch distribution for label-source, label-target and unlabel-target
> * whether the labeled target feature & score are used to update memory
> * the scheme to adjust weight $lambda$ of self-supervised loss term
> * whehter the hyperparameters are different for other datasets like DomainNet
> 
> BTW, I find the following code for NA classifier (Line 305 from demo_uda.py) seems not consistent with the Eq (6) in paper
> 
> `outputs_target = softmax_out**2 / ((softmax_out**2).sum(dim=0))`
> 
> From Eq (6) in the paper, the normalization is over prediction on all classes to make the distribution sharper, while in the code the parameter is `dim=0`, resulting normalization over all unlabeled data samples, will the normalization direction make big difference in final results ?
> 
> Appreciation if you can provide more detail about it~

Hi, Thanks for your interest. Since I am busy with some personal affairs, I will provide the code and details for  SSDA later.
As you mentioned, it seems that Line 305 is wrongly used for sharpening, I will correct that bug and update the new results later. Thanks a lot!

**Comment by tim-learn** [comment]
@lyxok1 Hi, I have corrected this typo in the new arxiv version and also uploaded the code corresponding to SSDA. You can try it now.
